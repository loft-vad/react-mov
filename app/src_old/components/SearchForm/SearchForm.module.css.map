{"version":3,"sourceRoot":"","sources":["SearchForm.module.scss"],"names":[],"mappings":"AAAA;EACE;EACA;EACA;;AAEA;EACE;EACA;EACA;EACA;EACA;EACA;EACA;EAEA;EACA;EACA;EACA;EAEA","file":"SearchForm.module.css"}