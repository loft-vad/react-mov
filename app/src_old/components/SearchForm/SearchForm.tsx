import React, { useState } from "react";
import styles from "./SearchForm.module.css";
import Button from "../common/Button/Button";

interface SearchProps {
  initialSearchString?: string;
  onSearch: (searchTerm: string) => void;
}

const SearchForm: React.FC<SearchProps> = ({ initialSearchString = "", onSearch }) => {
  const [searchString, setSearchString] = useState(initialSearchString);

  const handleKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
    if (event.key === "Enter") {
      onSearch(searchString);
    }
  };

  return (
    <div className={styles.searchWrapper}>
      <input
        type="text"
        placeholder="What do you want to watch?"
        onChange={(e) => setSearchString(e.target.value)}
        onKeyDown={handleKeyDown}
        value={searchString}
      />
      <Button text="Search" callBack={() => onSearch(searchString)} />
    </div>
  );
};

export default SearchForm;
