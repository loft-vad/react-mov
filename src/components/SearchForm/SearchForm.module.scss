.searchWrapper {
  display: flex;
  justify-content: space-between;
  gap: 14px;

  input {
    background: rgba(50, 50, 50, 0.8);
    mix-blend-mode: normal;
    opacity: 0.7;
    border-radius: 4px;
    width: 100%;
    border: none;
    padding: 17px 19px 16px;

    font-style: normal;
    font-weight: 400;
    font-size: 20px;
    line-height: 24px;

    color: #FFFFFF;
  }
}