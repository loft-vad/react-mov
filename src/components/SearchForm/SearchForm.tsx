import React, { useState } from 'react';
import styles from './SearchForm.module.scss';
import Button from '../common/Button/Button';

interface SearchProps {
  initialSearchString?: string;
}

const SearchForm: React.FC<SearchProps> = ({ initialSearchString = '' }) => {
  const [searchString, setSearchString] = useState(initialSearchString);

  const onSearch = () => {
    console.log('searchString ', searchString);
  };

  const handleKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
    if (event.key === 'Enter') {
      onSearch();
    }
  };

  return (
    <div className={styles.searchWrapper}>
      <input
        type='text'
        placeholder='What do you want to watch?'
        onChange={(e) => setSearchString(e.target.value)}
        onKeyDown={handleKeyDown}
        value={searchString}
      />
      <Button text='Search' callBack={() => onSearch()} />
    </div>
  );
};

export default SearchForm;
