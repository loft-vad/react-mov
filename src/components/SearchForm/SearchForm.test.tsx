import React from "react";

import SearchForm from "./SearchForm";
import { fireEvent, render } from "@testing-library/react";

const initialSearchString = "search string";

describe("SearchForm component", () => {
  it("should match snapshot", () => {
    const wrapper = render(<SearchForm onSearch={() => {}} />);

    expect(wrapper.container.firstChild).toMatchSnapshot();
  });
  it("renders an input with the value equal to initial value passed in props", () => {
    const wrapper = render(
      <SearchForm initialSearchString={initialSearchString} onSearch={() => {}} />,
    );

    expect(wrapper.container.firstChild).toMatchSnapshot();
  });
  it('after typing to the input and a "click" event on the Submit button, the "onChange" prop is called with proper value', () => {
    const onSearch = jest.fn();
    const wrapper = render(
      <SearchForm initialSearchString={initialSearchString} onSearch={onSearch} />,
    );
    const searchButton = wrapper.container.querySelector("button");

    expect(searchButton).toBeInTheDocument();
    fireEvent.click(searchButton!);
    expect(onSearch).toHaveBeenCalledWith(initialSearchString);
  });
  it('after typing to the input and pressing Enter key, the "onChange" prop is called with proper value', () => {
    const onSearch = jest.fn();
    const wrapper = render(
      <SearchForm initialSearchString={initialSearchString} onSearch={onSearch} />,
    );
    const searchInput = wrapper.container.querySelector("input");

    expect(searchInput).toBeInTheDocument();
    fireEvent.keyDown(searchInput!, { key: "Enter" });
    expect(onSearch).toHaveBeenCalledWith(initialSearchString);
  });
});
