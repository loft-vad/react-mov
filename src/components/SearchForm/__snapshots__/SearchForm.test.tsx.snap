// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm component renders an input with the value equal to initial value passed in props 1`] = `
<div
  class="searchWrapper"
>
  <input
    placeholder="What do you want to watch?"
    type="text"
    value="search string"
  />
  <button
    class="button"
    type="button"
  >
    Search
  </button>
</div>
`;

exports[`SearchForm component should match snapshot 1`] = `
<div
  class="searchWrapper"
>
  <input
    placeholder="What do you want to watch?"
    type="text"
    value=""
  />
  <button
    class="button"
    type="button"
  >
    Search
  </button>
</div>
`;
